Implemented Unit testing for SubscriberList and SubscriberDetail

elabcius5 opened the pull request:
- Implemented a small Unit Test to test the MVP of SubscriberList and SubscriberDetail
- Added variables of versions into gradle file

edwin-pixel-1 committed:
Implemented Unit test to SubsriberDetail MVP

- Added SubscriberDetailPresenterTest.java to test the MVP process of Subscriber Detail MVP

elabcius5 committed:
Implemented junit test for the app

elabcius5 committed:
Merge branch 'master' into juan/junit_testing

elabcius5 committed:
Implemented junit tests for SubriberList according Edwin indications

elabcius5 committed:
Merge branch 'Unit-Testing-2' into juan/junit_testing

elabcius5 committed:
Removed unnecesary variables into gradle

elabcius5 committed:
Added new variables into gradle file

lucaslabs commented on build.gradle:
```diff
@@ -21,3 +21,17 @@ allprojects {
+
+ext {
+   
+    minSdkVersion = 14
```
Nice!

juliocatano replied:
+1

lucaslabs commented on app/build.gradle:
```diff
@@ -21,12 +21,15 @@ android {
-    compile 'com.android.support:recyclerview-v7:23.1.1'
-    compile 'com.squareup.picasso:picasso:2.5.2'
-    compile 'com.android.support:cardview-v7:23.1.1'
+    compile "com.android.support:appcompat-v7:$rootProject.supportLibraryVersion"
```
Nice!

lucaslabs commented on app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java:
```diff
@@ -0,0 +1,106 @@
+import static org.mockito.Mockito.verify;
+
+/**
+ * @author edwin.cobos
```
Add javaDoc.

lucaslabs commented on app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java:
```diff
@@ -0,0 +1,106 @@
+
+    @Test
+    public void getUserInfoFromInteractorAndLoadIntoView() {
+
```
Add comments following the rule Given, When, Then.

lucaslabs commented on app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java:
```diff
@@ -0,0 +1,106 @@
+    @Test
+    public void getRepositoriesListFromInteractorAndLoadIntoView() {
+
+        Repository repo1 = new Repository(id_repo1, name_repo1, fullName_repo1, html_url_repo1);
```
Add commetns following the rull Given, When, Then.

lucaslabs commented on app/src/test/java/com/github/globant/githubsubscribers/subscriberslist/interactor/SubscribersListInteractorTest.java:
```diff
@@ -0,0 +1,62 @@
+    @Test
+    public void getSubcriberListFromInteractorAndLoadIntoView() {
+
+        List<Subscriber> subscriberList = new ArrayList<>(Arrays.asList(new Subscriber(login, id, avataUrl)));
```
Add commetns following the rull Given, When, Then.

lucaslabs commented on gradle.properties:
```diff
@@ -1,18 +1,19 @@
-# Project-wide Gradle settings.
```
Remove this file from commits adding a entrance on the `.gitignore` file.

juliocatano replied:
Normally you don't remove this file because it can contain app version (in case of library) or connection info for jenkins build. I don't see the harm on having this file in here because it doesn't change

elabcius5 committed:
Fixed tests according Lucas comments

lucaslabs commented on app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java:
```diff
@@ -70,36 +74,49 @@
+        //Then User information is loaded from Interactor and the callback is captured
         verify(interactor).getUserData(eq(USER_NAME_PARAM), OnFinishedListenerCallBack.capture());
 
+        //When User information is finally loaded
```
You can use When only once. Remove this comment.

lucaslabs commented on app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java:
```diff
@@ -70,36 +74,49 @@
+        //When User information is finally loaded
         OnFinishedListenerCallBack.getValue().onFinishedUser(user);
 
+        //Then User information is shown in UI
```
You can use Then only once. Remove this comment.

lucaslabs commented on app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java:
```diff
@@ -70,36 +74,49 @@
+        //Then Repositories list is loaded from Interactor and the callback is captured
         verify(interactor).getUserRepositoryData(eq(USER_NAME_PARAM), OnFinishedListenerCallBack.capture());
 
+        //When Repositories list is finally loaded
```
You can use When only once. Remove this comment.

lucaslabs commented on app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java:
```diff
@@ -70,36 +74,49 @@
+        //When Repositories list is finally loaded
         OnFinishedListenerCallBack.getValue().onFinishedRepository(repoList);
 
+        //Then Repositories list is shown in UI
```
You can use Then only once. Remove this comment.

lucaslabs commented on app/src/test/java/com/github/globant/githubsubscribers/subscriberslist/SubscribersListInteractorTest.java:
```diff
@@ -45,17 +51,26 @@
+        //Then Subscribers list is loaded from Interactor and the callback is captured
         verify(interactor).getSubscribersDataList(OnFinishedListenerCallBack.capture());
+
+        //When Subscribers list is finally loaded
```
You can use When only once. Remove this comment.

lucaslabs commented on app/src/test/java/com/github/globant/githubsubscribers/subscriberslist/SubscribersListInteractorTest.java:
```diff
@@ -45,17 +51,26 @@
+        //When Subscribers list is finally loaded
         OnFinishedListenerCallBack.getValue().onResponse(subscriberList);
+
+        //Then Subscribers list is shown in UI
```
You can use Then only once. Remove this comment.

elabcius5 committed:
Removed gradle.properties file

elabcius5 committed:
Removed When and Then duplicated comments

lucaslabs commented:
LG

elabcius5 committed:
Implemented a new Unit test for SubscribersList and added a dependency for PowerMockito

edwin-pixel-1 committed:
Added Unit tests to try error cases

edwin-pixel-1 committed:
Merge branch 'juan/junit_testing' of https://github.com/alejo740/GithubSubscribers into juan/junit_testing

juliocatano commented on .gitignore:
```diff
@@ -16,6 +16,7 @@ out/
 # Gradle files
 .gradle/
 build/
+gradle.properties
```
should you ignore the changes in the `gradle.properties`?

juliocatano commented on app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java:
```diff
@@ -0,0 +1,225 @@
+    private static String avatar_url = "https://avatars.githubusercontent.com/u/336005?v=3";
+    private static String html_url = "https://github.com/keyboardsurfer";
+    private static String name = "Ben Weiss";
+    private static String company = "@Google ";
```
should it have and space after the `@Google`?

edwin-pixel-1 replied:
Done!

edwin-pixel-1 committed:
Removed space in fake location variable

- Removed space at the end of fake 'location' variable in SubscriberDetailPresenterTest

juliocatano commented:
LG

lucaslabs commented on app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java:
```diff
@@ -0,0 +1,225 @@
+
+        OnFinishedListenerCallBack.getValue().onFinishedUser(user);
+
+        view.showSubscriberDetails(user);
```
Yo need to call `verify(view).method(param)`. Please refer to https://codelabs.developers.google.com/codelabs/android-testing/index.html?index=..%2F..%2Findex#5

lucaslabs commented on app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java:
```diff
@@ -0,0 +1,225 @@
+        OnFinishedListenerCallBack.getValue().onFailureUser(errorMessage, typeError);
+
+        int messageId = ErrorMessagesHelper.getMessage(typeError);
+        view.showUserError(messageId);
```
Yo need to call `verify(view).method(param)`. Please refer to https://codelabs.developers.google.com/codelabs/android-testing/index.html?index=..%2F..%2Findex#5

lucaslabs commented on app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java:
```diff
@@ -0,0 +1,225 @@
+        OnFinishedListenerCallBack.getValue().onFailureUser(errorMessage, typeError);
+
+        int messageId = ErrorMessagesHelper.getMessage(typeError);
+        view.showUserError(messageId);
```
Yo need to call `verify(view).method(param)`. Please refer to https://codelabs.developers.google.com/codelabs/android-testing/index.html?index=..%2F..%2Findex#5

lucaslabs commented on app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java:
```diff
@@ -0,0 +1,225 @@
+
+        OnFinishedListenerCallBack.getValue().onFinishedRepository(repoList);
+
+        view.showSubscriberUserRepositories(repoList);
```
Yo need to call `verify(view).method(param)`. Please refer to https://codelabs.developers.google.com/codelabs/android-testing/index.html?index=..%2F..%2Findex#5

lucaslabs commented on app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java:
```diff
@@ -0,0 +1,225 @@
+        OnFinishedListenerCallBack.getValue().onFailureRepository(errorMessage, typeError);
+
+        int messageId = ErrorMessagesHelper.getMessage(typeError);
+        view.showUserError(messageId);
```
Yo need to call `verify(view).method(param)`. Please refer to https://codelabs.developers.google.com/codelabs/android-testing/index.html?index=..%2F..%2Findex#5

lucaslabs commented on app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java:
```diff
@@ -0,0 +1,225 @@
+        OnFinishedListenerCallBack.getValue().onFailureRepository(errorMessage, typeError);
+
+        int messageId = ErrorMessagesHelper.getMessage(typeError);
+        view.showUserError(messageId);
```
Yo need to call `verify(view).method(param)`. Please refer to https://codelabs.developers.google.com/codelabs/android-testing/index.html?index=..%2F..%2Findex#5

lucaslabs commented on app/src/test/java/com/github/globant/githubsubscribers/subscriberslist/SubscribersListInteractorTest.java:
```diff
@@ -0,0 +1,141 @@
+
+        OnFinishedListenerCallBack.getValue().onResponse(subscriberList);
+
+        view.showSubscribersList(subscriberList);
```
Yo need to call `verify(view).method(param)`. Please refer to https://codelabs.developers.google.com/codelabs/android-testing/index.html?index=..%2F..%2Findex#5

lucaslabs commented on app/src/test/java/com/github/globant/githubsubscribers/subscriberslist/SubscribersListInteractorTest.java:
```diff
@@ -0,0 +1,141 @@
+        OnFinishedListenerCallBack.getValue().onFailure(errorMessage, typeError);
+
+        int messageId = ErrorMessagesHelper.getMessage(typeError);
+        view.showSubscribersError(messageId);
```
Yo need to call `verify(view).method(param)`. Please refer to https://codelabs.developers.google.com/codelabs/android-testing/index.html?index=..%2F..%2Findex#5

lucaslabs commented on app/src/test/java/com/github/globant/githubsubscribers/subscriberslist/SubscribersListInteractorTest.java:
```diff
@@ -0,0 +1,141 @@
+        OnFinishedListenerCallBack.getValue().onFailure(errorMessage, typeError);
+
+        int messageId = ErrorMessagesHelper.getMessage(typeError);
+        view.showSubscribersError(messageId);
```
Yo need to call `verify(view).method(param)`. Please refer to https://codelabs.developers.google.com/codelabs/android-testing/index.html?index=..%2F..%2Findex#5
